truefoundry.yaml

```diff
@@ -0,0 +1,403 @@
+      build_source:
+        # ref: b2c1a10d37d19f80f0473bd28832dfb426deb050
+        # type: git
+        # repo_url: https://github.com/truefoundry/cognita
+        # branch_name: ps_cleanup_metadatastore
+        local_build: false
+        type: local
+    ports:
+      - host: cas.truefoundry.com
+        port: 5000
+        expose: true
+        protocol: TCP
+        app_protocol: http
+    replicas: 1
+    resources:
+      cpu_limit: 0.1
+      cpu_request: 0.05
+      memory_limit: 200
+      memory_request: 100
+      ephemeral_storage_limit: 200
+      ephemeral_storage_request: 100
+  - name: cas-qdrant
+    type: helm
+    source:
+      type: helm-repo
+      chart: qdrant
+      version: 0.8.4
+      repo_url: https://qdrant.github.io/qdrant-helm
+    values:
+      service:
+        type: ClusterIP
+        ports:
+          - name: http
+            port: 6333
+            protocol: TCP
+            targetPort: 6333
+            checksEnabled: true
+          - name: grpc
+            port: 6334
+            protocol: TCP
+            targetPort: 6334
+            checksEnabled: false
+          - name: http-p2p
+            port: 6335
+            protocol: TCP
+            targetPort: 6335
+            checksEnabled: false
+      persistence:
+        size: 50G
+      tolerations:
+        - key: kubernetes.azure.com/scalesetpriority
+          value: spot
+          effect: NoSchedule
+          operator: Equal
+        - key: cloud.google.com/gke-spot
+          value: "true"
+          effect: NoSchedule
+          operator: Equal
+      replicaCount: 2
+      fullnameOverride: cas-qdrant
+    kustomize:
+      additions:
+        - kind: VirtualService
+          spec:
+            http:
+              - match:
+                  - uri:
+                      prefix: /qdrant/
+                route:
+                  - destination:
+                      host: cas-qdrant.cognita-internal.svc.cluster.local
+                      port:
+                        number: 6333
+                rewrite:
+                  uri: /
+              - match:
+                  - headers:
+                      x-route-service:
+                        exact: qdrant-ui
+                route:
+                  - destination:
+                      host: cas-qdrant.cognita-internal.svc.cluster.local
+                      port:
+                        number: 6333
+                rewrite:
+                  uri: /
+            hosts:
+              - cas-qdrant-ui.truefoundry.com
+            gateways:
+              - istio-system/tfy-wildcard
+          metadata:
+            name: cas-qdrant
+            namespace: cognita-internal
+          apiVersion: networking.istio.io/v1alpha3
+
+  # Qdrant-UI
+  - name: cas-qdrant-ui
+    type: service
+    image:
+      type: build
+      build_source:
+        type: git
+        repo_url: https://github.com/truefoundry/qdrant-web-ui-new
+        branch_name: support-path-based-routing
```

Comment:
Last I remember there was some header based hack we had to do to make path based routing work. 
I think in general it would be better we use `path` for all services, that way we ensure path based routing is always tested